narwhal/executor/src/subscriber.rs

```diff
@@ -231,8 +233,38 @@ impl<Network: SubscriberNetwork> Fetcher<Network> {
             batches: Vec::with_capacity(num_certs),
         };
 
+        let mut workers_for_certificates: HashSet<NetworkPublicKey> = HashSet::new();
+        let mut batch_digests: HashMap<WorkerId, HashSet<BatchDigest>> = HashMap::new();
+
+        let mut num_batches = 0;
         for cert in &sub_dag.certificates {
-            let mut batches = Vec::with_capacity(num_batches);
+            for (digest, (worker_id, _)) in cert.header.payload.iter() {
+                let workers = self.network.workers_for_certificate(cert, worker_id);
```

Comment:
I don't think we need to collect workers from the certificates. Instead, using all workers in the network should be better.

Reply:
Workers from the CommitttedSubdag may approximate all workers, so my preference is to keep things simple. But I don't mind keeping the current logic.